### PR TITLE
Propose constitution with guardrail script

### PR DESCRIPTION
**Pull request by Jimbo4350**
# Description

Add your description here, if it fixes a particular issue please provide a
[link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)
to the issue.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -32,8 +32,8 @@ packages:
   trace-resources
   trace-forward
 
-program-options
-  ghc-options: -Werror
+-- program-options
```
Comment (carbolymer) [inline cabal.project:35]: This is useful in CI.

**Review by carbolymer — COMMENTED** [review]
Isn't that a duplicate of https://github.com/IntersectMBO/cardano-node/pull/5685?

**Comment by Jimbo4350** [comment]
> Isn't that a duplicate of #5685?

No, this puts a guard rail script in the constitution.

**Review by carbolymer — APPROVED** [review]

File: cardano-testnet/cardano-testnet.cabal [hunk 2]
```diff
@@ -198,8 +198,10 @@ test-suite cardano-testnet-test
                       , cardano-cli
                       , cardano-crypto-class
                       , cardano-ledger-conway
+                      , cardano-ledger-core
                       , cardano-ledger-shelley
                       , cardano-node
+                      , cardano-strict-containers
```
Comment (smelc) [inline cardano-testnet/cardano-testnet.cabal:204]:
```suggestion
                      , cardano-strict-containers ^>= 0.1
```

To have the same bound as [the existing occurrence](https://github.com/IntersectMBO/cardano-node/blob/master/bench/locli/locli.cabal#L146) in this repo?

**Pull request merged**